src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.NonAndroid.cs

```diff
@@ -403,6 +413,11 @@ private static bool CompareTimeZoneFile(string filePath, byte[] buffer, byte[] r
         /// </summary>
         private static string FindTimeZoneId(byte[] rawData)
         {
+            if (Invariant)
```

Comment:
>Invariant [](http://example.com/codeflow?start=16&length=9)

Is it possible to reach here when Invariant is true? I think if not, we should assert in all such places.